src/plugins/teleport-plugin-react-base-component/utils.ts

```diff
@@ -124,6 +125,38 @@ export const generateConditionalNode = (
   return createConditionalJSXExpression(subTree, condition, conditionIdentifier)
 }
 
+export const generateSlotNode = (node: UIDLSlotNode, accumulators: ReactComponentAccumulators) => {
+  const childrenProp: UIDLDynamicReference = {
+    type: 'dynamic',
+    content: {
+      referenceType: 'prop',
+      id: 'children',
+    },
+  }
+
+  const childrenExpression = makeDynamicValueExpression(childrenProp)
+
+  if (node.content.fallback) {
+    const fallbackContent = generateNodeSyntax(node.content.fallback, accumulators)
+    let expression: types.Expression
+
+    if (typeof fallbackContent === 'string') {
+      expression = types.stringLiteral(fallbackContent)
+    } else if ((fallbackContent as types.JSXExpressionContainer).expression) {
+      expression = (fallbackContent as types.JSXExpressionContainer).expression as types.Expression
+    } else {
+      expression = fallbackContent as types.JSXElement
+    }
+
+    // props.children with fallback
+    return types.jsxExpressionContainer(
```

Comment:
are we drooping the optional last param `t=types` form the functions now? If yes why, and should we remove all the optional params t=types now?

Reply:
no, I think we can keep it, I missed that here, I'll add it. but maybe we should isolate at some point babel/types usage to some specific utils/builders modules so we can test the other layers without worrying about this.